Fake handcuffs no longer apply click cooldown on resist

Rhials opened the pull request:
## About The Pull Request

Fake handcuffs/zipties do not apply a click cooldown upon resisting.

I'm not super happy with the solution to this (adding another item-level var) but I told myself I'd fix this, and wasn't about to give up after my handcuff-level solution didn't work.

Which calls into question -- Why does the breakout proc use item typecasting over the restraint type? Is this necessary for anything to function? I was too afraid to change it and accidentally break something else.
## Why It's Good For The Game

Closes #82711.

Makes fake handcuffs a properly harmless joke item.
## Changelog
:cl: Rhials
fix: Fake handcuffs/Zipties no longer block clicks for a few seconds after being resisted out of.
/:cl:

Rhials committed:
wat

MrMelbert commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -276,7 +276,7 @@
 		type = 2
 	if(I)
-		if(type == 1)
+		if(type == 1 && I.breakout_cooldown)
```
Rather than outright not applying the CD, the duration of the CD itself should be variable, ie `var/resist_cd = CLICK_CD_BREAKOUT`.

These should still have SOME cd associated, even if quick (like 1 second)

Rhials committed:
addresses comment

fake cuffs are taking a long time to break out of and I need to confirm this is not my fault

itseasytosee commented:
No offense intended, but this is indeed a bandaid. TG's click cooldown system as of current is in need of a refactoring, if not a reimagining.
